### PR TITLE
[BE] Search product with name

### PR DESCRIPTION
**Pull request by DTHNhung**
## Related Tickets:

-   [#46851](https://edu-redmine.sun-asterisk.vn/issues/46851)

## WHAT this PR does?

-  Back-end Search product - name

### CODE REVIEW

**Review by khieutrung12 — APPROVED** [review]

**Comment by DTHNhung** [comment]
ready

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -15,4 +17,41 @@ public function index()
     {
         return view('home');
     }
+
+    public function search(Request $request)
+    {
+        $products = Product::where('name', 'LIKE', '%' . $request->name . '%')
+            ->skip(0)->take(config('app.limit_search'))->get();
```
Comment (huyenttm-0901) [inline app/Http/Controllers/HomeController.php]: skip(0) làm gì v em?

File: app/Http/Controllers/HomeController.php [hunk 2]
```diff
@@ -15,4 +17,41 @@ public function index()
...
+            return view('error_search');
+        }
+
+        $brands = [];
+        foreach ($products as $product) {
+            if (isset($brands[$product->brand->name])) {
+                $brands[$product->brand->name] += 1;
+            } else {
+                $brands[$product->brand->name] = 1;
+            }
```
Comment (huyenttm-0901) [inline app/Http/Controllers/HomeController.php:48]: n+1 queries

**Comment by DTHNhung** [comment]
ready

**Pull request merged**